### ISSUE TITLE
escape-case and no-hex-escape false positives – need to handle backslashes

### ISSUE THREAD

**Issue by lydell** [sindresorhus/eslint-plugin-unicorn#85]
package.json:

```json
{
  "private": true,
  "eslintConfig": {
    "env": {
      "es6": true
    },
    "parserOptions": {
      "ecmaVersion": 2017,
      "sourceType": "module"
    },
    "plugins": [
      "unicorn"
    ],
    "rules": {
      "unicorn/escape-case": "error",
      "unicorn/no-hex-escape": "error"
    }
  },
  "scripts": {
    "test": "eslint test.js"
  },
  "devDependencies": {
    "eslint": "^3.15.0",
    "eslint-plugin-unicorn": "^2.0.1"
  }
}
```

test.js:

```js
"\\ufeff"
"\\\\xa0"
```

Output:

```
$ npm test

> @ test /home/lydell/bugs/eslint-plugin-unicorn-bugs
> eslint test.js


/home/lydell/bugs/eslint-plugin-unicorn-bugs/test.js
  1:1  error  Use uppercase characters for the value of the escape sequence  unicorn/escape-case
  2:1  error  Use uppercase characters for the value of the escape sequence  unicorn/escape-case
  2:1  error  Use unicode escapes instead of hexadecimal escapes             unicorn/no-hex-escape

✖ 3 problems (3 errors, 0 warnings)

npm ERR! Test failed.  See above for more details.
```

Expected: No errors.

**Comment by sindresorhus** [comment]
// @Bebersohl In case you have interest in looking into this

**Comment by sindresorhus** [comment]
I just encountered this myself. I had `internal\\child_process` which got reported and fixed as `internal\\cHild_process`...

**Comment by sindresorhus** [comment]
// @florianb In case you want to take a look at this ;)

**Comment by sindresorhus** [comment]
// @matijs

**Comment by florianb** [comment]
Unless somebody else is taking it, i will dig in at monday.. 😊

**Comment by matijs** [comment]
@florianb I'd like to fix no-hex-escape if that's alright with you :)

**Comment by florianb** [comment]
You are welcome! ;) Happy hacking and thanks!
Matijs Brinkhuis <notifications@github.com> schrieb am 26.03.2017 12:46 nachm.:

@florianb<https://github.com/florianb> I'd like to fix no-hex-escape if that's alright with you :)
